test/unit/calculators/checks_benefits_support_calculator_test.rb

```diff
@@ -221,473 +335,709 @@ class CheckBenefitsSupportCalculatorTest < ActiveSupport::TestCase
+          end
 
+          should "be false if country is NI, under state pension age and over 16000 assets" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.where_do_you_live = "northern-ireland"
             calculator.over_state_pension_age = "no"
             calculator.assets_and_savings = "over_16000"
             assert_not calculator.eligible_for_universal_credit_ni?
           end
-
-          calculator.where_do_you_live = "northern-ireland"
-          calculator.over_state_pension_age = "yes"
-          assert_not calculator.eligible_for_universal_credit_ni?
-
-          calculator.where_do_you_live = "northern-ireland"
-          calculator.over_state_pension_age = "no"
-          calculator.assets_and_savings = "over_16000"
-          assert_not calculator.eligible_for_universal_credit_ni?
         end
       end
 
       context "#eligible_for_housing_benefit?" do
-        should "return true if eligible for Housing Benefit" do
-          calculator = CheckBenefitsSupportCalculator.new
-          %w[england wales].each do |country|
-            calculator.where_do_you_live = country
-            calculator.over_state_pension_age = "yes"
-            assert calculator.eligible_for_housing_benefit?
+        context "when eligible" do
+          should "be true if country is England or Wales and over state pension age" do
+            %w[england wales].each do |country|
+              calculator = CheckBenefitsSupportCalculator.new
+              calculator.where_do_you_live = country
+              calculator.over_state_pension_age = "yes"
+              assert calculator.eligible_for_housing_benefit?
+            end
           end
         end
 
-        should "return false if not eligible for Housing Benefit" do
-          calculator = CheckBenefitsSupportCalculator.new
-          %w[scotland northern-ireland].each do |country|
-            calculator.where_do_you_live = country
-            calculator.over_state_pension_age = "no"
-            assert_not calculator.eligible_for_housing_benefit?
+        context "when ineligible" do
+          should "be false if country is England or Wales and UNDER state pension age" do
+            calculator = CheckBenefitsSupportCalculator.new
+            %w[england wales].each do |country|
+              calculator = CheckBenefitsSupportCalculator.new
+              calculator.where_do_you_live = country
+              calculator.over_state_pension_age = "no"
+              assert_not calculator.eligible_for_housing_benefit?
+            end
+          end
+
+          should "be false if country is not England or Wales and over state pension age" do
+            calculator = CheckBenefitsSupportCalculator.new
+            %w[scotland northern-ireland].each do |country|
+              calculator = CheckBenefitsSupportCalculator.new
+              calculator.where_do_you_live = country
+              calculator.over_state_pension_age = "no"
```

Comment:
Should be calculator.over_state_pension_age = "yes"

Reply:
oh wait that was right! The test description needs changing to 'be false if country is not England or Wales and under state pension age'